### PR TITLE
Read multiple directories to construct Python ledger

### PR DESCRIPTION
**Pull request by eddyashton**
Fix for Daily CI. When a node's ledger is split across multiple directories, we should read from all of them to reconstruct the full ledger.

### CODE REVIEW

File: python/ccf/ledger.py [hunk 1]
```diff
@@ -496,24 +496,29 @@ def _reset_iterators(self):
...
-    def __init__(self, directory: str):
+    def __init__(self, directories: List[str]):
 
         self._filenames = []
 
-        ledgers = os.listdir(directory)
+        ledger_files = []
+        for directory in directories:
+            for path in os.listdir(directory):
+                chunk = os.path.join(directory, path)
```
Comment (jumaffre) [inline python/ccf/ledger.py:506]: Should we be defensive here and check that `path` hasn't already been read, e.g. if the same directory is passed twice, or if the same ledger chunk (based on its name only) exists in two directories?

**Review by jumaffre — APPROVED** [review]

File: python/ccf/ledger.py (hunk 1 above)
Comment (achamayou) [reply python/ccf/ledger.py:506]: If the same chunk occurs twice, there will be a validation error in the signatures. Properly making sure there are no duplicates (ie. resolving symlinks etc) may be more trouble than it's worth in practice.

**Review by achamayou — APPROVED** [review]

File: python/ccf/ledger.py (hunk 1 above)
Comment (jumaffre) [reply python/ccf/ledger.py:506]: Fair point. I thought we could easily prevent suspicious signature verification errors here, but resolving symlinks, etc. seems too convoluted indeed.

Comment (achamayou) [reply python/ccf/ledger.py:506]: I suspect we can use the parsing of the sequence numbers to implement an easy but still helpful check.

**Pull request merged**